File: Data Structures & Algorithms/max-water-container/submission-0.java
class Solution {
    public int maxArea(int[] heights) {
        int res = , l = 0, r = heights.length - 1;
        while (l < r){
            int amount = Math.min(heights[l], heights[r]) * (r-l);
            res = Math.max(res, amount);

            if (heights[l] < heights[r]) l++;
            else r--;
        }
        return res;
    }
}

File: Data Structures & Algorithms/eating-bananas/submission-0.py
class Solution:
    def minEatingSpeed(self, piles: List[int], h: int) -> int:
        paces = [i for i in range(1, max(piles))]

        l, r = 0, len(paces) - 1
        res = max(piles)

        while l <= r:
            m = l + (r - l) // 2

            pace = paces[m]
            time = 0

            for pile in piles:
                time += math.ceil(pile / pace)
            
            if time <= h:
                res = min(res, pace)
                r = m - 1
            else:
                l = m + 1
        return res